libtransmission/CMakeLists.txt

```diff
@@ -305,6 +306,10 @@ target_link_libraries(${TR_NAME}
         small::small
         libevent::event)
 
+if(NOT HAVE_CXX_ATOMICS_WITHOUT_LIB)
```

Comment:
Avoid extra conditional blocks if possible, do as above instead, i.e. add `$<$<BOOL:${HAVE_LIBATOMIC}>:atomic>` to the existing `target_link_libraries()` call